hil-test/tests/clock_monitor.rs

```diff
@@ -0,0 +1,52 @@
+//! Clock Monitor Test
+
+//% CHIPS: esp32 esp32c2 esp32c3 esp32c6 esp32h2 esp32s2 esp32s3
+
+#![no_std]
+#![no_main]
+
+use defmt_rtt as _;
+use esp_backtrace as _;
+use esp_hal::{clock::ClockControl, peripherals::Peripherals, prelude::*, rtc_cntl::Rtc};
+
+struct Context<'a> {
+    rtc: Rtc<'a>,
+}
+
+impl Context<'_> {
+    pub fn init() -> Self {
+        let peripherals = Peripherals::take();
+        let system = peripherals.SYSTEM.split();
+        ClockControl::boot_defaults(system.clock_control).freeze();
+
+        let rtc = Rtc::new(peripherals.LPWR, None);
+
+        Context { rtc }
+    }
+}
+
+#[cfg(test)]
+#[embedded_test::tests]
+mod tests {
+    use super::*;
+
+    #[init]
+    fn init() -> Context<'static> {
+        Context::init()
+    }
+
+    #[test]
+    fn test_estimated_clock(mut ctx: Context<'static>) {
+        // We call the function twice since sometimes the first call gives a
+        // wrong result
+        ctx.rtc.estimate_xtal_frequency();
+        let estimated_xtal_freq = ctx.rtc.estimate_xtal_frequency();
+
+        #[cfg(feature = "esp32c2")] // 26 MHz
+        assert!(estimated_xtal_freq > 22 && estimated_xtal_freq < 30);
```

Comment:
Regarding these ranges, +- 4mhz each side is quite a bit, can it really vary this much or are we playing to safe?

Reply:
I was cautious since I dont know how good the estimate is. I've seen values of 38 for 40MHz freqs and 30 for 32 MHz freqs. Shall we make it +-3?

Reply:
Okay let's try +-3!

Reply:
I think we need to broaden our range, as Im getting 35 MHz in a C6.